src/Factories/Transform/UserSchema/OptionalIfTransformFactory.cs

```diff
@@ -0,0 +1,38 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using form_builder.Conditions;
+using form_builder.Models;
+using form_builder.Models.Elements;
+
+namespace form_builder.Factories.Transform.UserSchema
+{
+    public class OptionalIfTransformFactory : IUserPageTransformFactory
+    {
+        public async Task<Page> Transform(Page page, FormAnswers convertedAnswers)
+        { 
+            IEnumerable<IElement> elements = page.ValidatableElements
+                .Where(element => !string.IsNullOrEmpty(element.Properties.OptionalIf.QuestionId));
+
+            if (!elements.Any())
+                return await Task.FromResult(page);
+
+            foreach (IElement element in elements)
+            {
+                Answers answers = convertedAnswers.AllAnswers
+                    .FirstOrDefault(answer => answer.QuestionId
+                    .Equals(element.Properties.OptionalIf.QuestionId));
+
+                var questionsAnswers = new Dictionary<string, dynamic>
```

Comment:
Just to follow the coding convention within this class. Replace var with Type. example Dictionary<string, dynamic> questionsAnswers = new()